x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/GetMigrationReindexStatusTransportAction.java

```diff
@@ -96,11 +111,91 @@ void getRunningTaskFromNode(String persistentTaskId, ActionListener<Response> li
+            ReindexDataStreamStatus status = (ReindexDataStreamStatus) info.status();
+            Set<String> inProgressIndices = status.inProgress();
+            if (inProgressIndices.isEmpty()) {
+                // We have no reason to fetch index stats since there are no in progress indices
+                reportStatus(Map.of(), status, listener);
+            } else {
+                fetchInProgressStatsAndReportStatus(inProgressIndices, status, listener);
+            }
         }
     }
 
-    private void runOnNodeWithTaskIfPossible(Task thisTask, Request request, String nodeId, ActionListener<Response> listener) {
+    /*
+     * The status is enriched with the information from inProgressMap to create a new ReindexDataStreamEnrichedStatus, which is used in the
+     * response sent to the listener.
+     */
+    private void reportStatus(
+        Map<String, Tuple<Long, Long>> inProgressMap,
+        ReindexDataStreamStatus status,
+        ActionListener<Response> listener
+    ) {
+        ReindexDataStreamEnrichedStatus enrichedStatus = new ReindexDataStreamEnrichedStatus(
+            status.persistentTaskStartTime(),
+            status.totalIndices(),
+            status.totalIndicesToBeUpgraded(),
+            status.complete(),
+            status.exception(),
+            inProgressMap,
+            status.pending(),
+            status.errors()
+        );
+        listener.onResponse(new Response(enrichedStatus));
+    }
+
+    /*
+     * This method feches doc counts for all indices in inProgressIndices (and the indices they are being reindexed into). After
+     * successfully fetching those, reportStatus is called.
+     */
+    private void fetchInProgressStatsAndReportStatus(
+        Set<String> inProgressIndices,
+        ReindexDataStreamStatus status,
+        ActionListener<Response> listener
+    ) {
+        IndicesStatsRequest indicesStatsRequest = new IndicesStatsRequest();
+        String[] indices = inProgressIndices.stream()
+            .flatMap(index -> Stream.of(index, ReindexDataStreamIndexTransportAction.generateDestIndexName(index)))
+            .toList()
+            .toArray(new String[0]);
+        indicesStatsRequest.indices(indices);
+        /*
+         * It is possible that the destination index will not exist yet, so we want to ignore the fact that it is missing
+         */
+        indicesStatsRequest.indicesOptions(IndicesOptions.fromOptions(true, true, true, true));
+        client.execute(IndicesStatsAction.INSTANCE, indicesStatsRequest, new ActionListener<IndicesStatsResponse>() {
+            @Override
+            public void onResponse(IndicesStatsResponse indicesStatsResponse) {
+                Map<String, Tuple<Long, Long>> inProgressMap = new HashMap<>();
+                for (String index : inProgressIndices) {
+                    DocsStats totalDocsStats = indicesStatsResponse.getIndex(index).getTotal().getDocs();
```

Comment:
Could there a be race condition where `index` has already been deleted and  `indicesStatsResponse.getIndex(index)` is null?

Reply:
Oh, like if this were a rerun and we just hadn't deleted the old destination yet (since we do that at the time we handle an index rather than upfront)? Yeah I think you're probably right. It would be hard to run into, since we create the index right after we delete it (and you'd already have to be in a rerun and time it just right). But if it's possible, it'll happen eventually. I'll put in a null check.

Reply:
Oh, this is the source index that we're not deleting right now. But we might in the future. So I'll null check it just in case.

Reply:
Yeah, I was thinking about if we delete source. :+1: